elastic/apm-server
# processor/otel: add support for rpc conventions

## Motivation/summary

Handle `rpc.*` semantic conventions for OpenTelemetry:
 - `rpc.system` is used to identify RPC spans. The value
   is currently used for span subtype, and dropped for
   transactions as we have nowhere to store it.
 - `rpc.service` is dropped, as it is included in the
   operation name and we have nowhere to store it.
 - `rpc.method` is dropped, as it is included in the
   operation name and we have nowhere to store it.
 - `rpc.grpc.status_code` is used to set transaction result,
   but is dropped for spans.

Dropping fields is clearly not ideal. We should revisit
this and define an RPC schema which works for OpenTelemetry
as well as our own agents, and which fits into ECS.

RPC transactions are given the type "request", and spans
are given the type "external" with the `rpc.system` value
as the subtype.

`span.destination.service` is set from the peer address, like
we do for our own agents.

## Checklist


## How to test these changes

Instrument gRPC-based server and client with OpenTelemetry, collect traces with APM Server and check:
- transaction type should be "request"
- transaction result should be "OK" (or some other gRPC status code's string representation)
- span type should be "external", and subtype should be "grpc"
- client and server should be connected in the service map in Kibana
- there should be no "labels.rpc.*" fields

## Related issues

Closes https://github.com/elastic/apm-server/issues/4713
